service/lib/agama/storage/volume_templates_builder.rb

```diff
@@ -0,0 +1,198 @@
+          values[:filesystem] ||= Y2Storage::Filesystems::Type::EXT4
+
+          size = fetch(data, :size, {})
+          values[:auto_size] = fetch(size, :auto, false)
+          values[:min_size] = parse_disksize(fetch(size, :min))
+          values[:max_size] = parse_disksize(fetch(size, :max))
+        end
+      end
+
+      def btrfs(data)
+        btrfs_data = fetch(data, "btrfs", {})
+        BtrfsSettings.new.tap do |btrfs|
+          btrfs.snapshots = fetch(btrfs_data, "snapshots", false)
+          btrfs.read_only = fetch(btrfs_data, "read_only", false)
+          btrfs.default_subvolume = fetch(btrfs_data, "default_subvolume", "")
+          btrfs.subvolumes = fetch(btrfs_data, "subvolumes", []).map { |s| subvolume(s) }
+        end
+      end
+
+      def subvolume(data)
+        return Y2Storage::SubvolSpecification.new(data) if data.is_a?(String)
+
+        attrs = { copy_on_write: fetch(data, :copy_on_write), archs: fetch(data, :archs) }.compact
+        Y2Storage::SubvolSpecification.new(fetch(data, :path), **attrs)
+      end
+
+      def outline(data) # rubocop:disable Metrics/AbcSize
+        outline_data = fetch(data, "outline", {})
+        VolumeOutline.new.tap do |outline|
+          outline.required = fetch(outline_data, "required", false)
+          outline.filesystems = fetch(outline_data, "filesystems", [])
+          outline.filesystems.map! { |fs| fs_type(fs) }
+          outline.snapshots_configurable = fetch(outline_data, "snapshots_configurable", true)
+
+          size = fetch(outline_data, "auto_size", {})
+          min = parse_disksize(fetch(size, :base_min))
+          max = parse_disksize(fetch(size, :base_max))
+          outline.base_min_size = min if min
+          outline.base_max_size = max if max
+          outline.adjust_by_ram = fetch(size, :adjust_by_ram, false)
+          outline.min_size_fallback_for = Array(fetch(size, :min_fallback_for))
+          outline.min_size_fallback_for.map! { |p| cleanpath(p) }
+          outline.max_size_fallback_for = Array(fetch(size, :max_fallback_for))
+          outline.max_size_fallback_for.map! { |p| cleanpath(p) }
+
+          assign_snapshots_increment(outline, fetch(size, :snapshots_increment))
+        end
+      end
+
+      def assign_snapshots_increment(outline, increment)
+        return if increment.nil?
+
+        if increment =~ /(\d+)\s*%/
+          outline.snapshots_percentage = Regexp.last_match(1).to_i
+        else
+          outline.snapshots_size = Y2Storage::DiskSize.parse(increment, legacy_units: true)
+        end
+      end
+
+      def parse_disksize(value)
+        return nil unless value
+
+        Y2Storage::DiskSize.parse_or(value, nil, legacy_units: true)
+      end
+
+      def cleanpath(path)
+        Pathname.new(path).cleanpath.to_s
+      end
+
+      def fs_type(filesystem)
+        return filesystem if filesystem.is_a?(Y2Storage::Filesystems::Type)
+
+        Y2Storage::Filesystems::Type.find(filesystem.downcase.to_sym)
```

Comment:
to be honest I expect such method in that Type class instead of exposing that it is only lowercase here.

Reply:
Again, I would say this is part of the validation.

Reply:
Well, I am not sure if it is about validation that we allow to have also upper case fs names, but as change will need change in yast it is ok for me to have it as it is.